lib/ast/AST.cpp

```diff
@@ -692,7 +692,7 @@ void KORECompositePattern::prettyPrint(std::ostream &out, PrettyPrintData const&
           case '8':
           case '9': {
             std::string buf;
-            for(; i < format.length() && format[i] >= '0' && format[i] < '9'; i++) {
+            for(; i < format.length() && format[i] >= '0' && format[i] <= '9'; i++) {
```

Comment:
Do we only handle productions with arguments up to length 10?

Reply:
No. This was a bug only with numbers containing the number 9 in their decimal expansion.

Reply:
The fix should make it work for any arity, although there is another fix that will have to be included in the front end pr